audeer/core/io.py

```diff
@@ -211,21 +212,31 @@ def create_archive(
     )
     disable = not verbose
 
-    if archive.endswith("zip"):
-        with zipfile.ZipFile(archive, "w", zipfile.ZIP_DEFLATED) as zf:
-            for file in progress_bar(files, desc=desc, disable=disable):
-                full_file = safe_path(root, file)
-                zf.write(full_file, arcname=file)
-    elif archive.endswith("tar.gz"):
-        with tarfile.open(archive, "w:gz") as tf:
-            for file in progress_bar(files, desc=desc, disable=disable):
-                full_file = safe_path(root, file)
-                tf.add(full_file, file)
-    else:
+    archive_handlers = {
```

Comment:
My understanding is that these archive handlers are wisely chosen:
After doing some research I found

- that "tgz" and "tbz" are only there for historical reasons - with windows only wanting a single extension of length 3. 

- 7-zip has high compression rate it is said to be slow, has external dependencies that might not be installed. 

So I think that it is fully ok to not provide these by default.
 
What I wonder though: in case an impatient user needs to patch an archive handler for a new compression that `audeer` cannot integrate fast enough: would it make sense to move these handlers into something like a  `define`  module such that the handler can be patched in an ad-hoc fashion by the impatient?

Reply:
Would be great if it would work that way, but I'm afraid that it is not enough to define how to open the file as even the existing archive handlers have already a different syntax to add a file to the archive:

```python
            if extension == "zip":
                archive_file.write(full_file, arcname=file)
            else:
                archive_file.add(full_file, file)
```

So I guess to make it easier extendable, one would need to introduce something like a base classe.

I would propose to stay here with the current solution, but maybe open an issue requesting to make it easier to integrate custom or other archive handlers.

Reply:
Yes, sure, `gzip` and `bzip` do not handle archiving this is why they resort to `tar`  for doing the achieving, this is why it is being called  `add` as opposed to `write` in zip.

I would not oppose having a base class or equivalent mechanism to handle adding/writing.
But the argument that this is beyond scope here is enough to suggest that we can  resolve here.